pkg/controller/operators/catalog/operator.go

```diff
@@ -551,6 +554,66 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 					plan.Status.Plan[i].Status = v1alpha1.StepStatusCreated
 				}
 
+			case roleKind:
+				// Marshal the manifest into a Role instance.
+				var r rbacv1.Role
+				err := json.Unmarshal([]byte(step.Resource.Manifest), &r)
+				if err != nil {
+					return err
+				}
+
+				// Attempt to create the Role.
+				_, err = o.OpClient.KubernetesInterface().RbacV1().Roles(plan.Namespace).Create(&r)
+				if k8serrors.IsAlreadyExists(err) {
+					// If it already existed, mark the step as Present.
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusPresent
+				} else if err != nil {
+					return err
+				} else {
+					// If no error occurred, mark the step as Created.
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusCreated
+				}
+
+			case roleBindingKind:
+				// Marshal the manifest into a RoleBinding instance.
+				var rb rbacv1.RoleBinding
+				err := json.Unmarshal([]byte(step.Resource.Manifest), &rb)
+				if err != nil {
+					return err
+				}
+
+				// Attempt to create the RoleBinding.
+				_, err = o.OpClient.KubernetesInterface().RbacV1().RoleBindings(plan.Namespace).Create(&rb)
+				if k8serrors.IsAlreadyExists(err) {
+					// If it already existed, mark the step as Present.
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusPresent
+				} else if err != nil {
+					return err
+				} else {
+					// If no error occurred, mark the step as Created.
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusCreated
+				}
+
+			case serviceAccountKind:
+				// Marshal the manifest into a Role instance.
+				var sa corev1.ServiceAccount
+				err := json.Unmarshal([]byte(step.Resource.Manifest), &sa)
+				if err != nil {
+					return err
+				}
+
+				// Attempt to create the Role.
+				_, err = o.OpClient.KubernetesInterface().CoreV1().ServiceAccounts(plan.Namespace).Create(&sa)
+				if k8serrors.IsAlreadyExists(err) {
```

Comment:
I think we need more here.

Suppose I have CSV1 already in my cluster. I create an installplan for CSV2, and CSV2 replaces CSV1, and that they both use the same serviceAccountName in their permission block (a fairly common scenario with our existing set of CSVs)

Applying the installplan for CSV2 will check if the serviceaccount exists, see that it does, and mark it present. But the ownerrefs on the serviceaccount will still point to CSV1. When CSV2 gets applied to the cluster it will kick off the upgrade and delete CSV1, which will clean up the serviceaccount that CSV2 relied upon.

This might make more sense to solve in the CSV upgrade code (adopt the serviceaccount), but one way to do it would be to have it add or update the ownerrefs in this step.